### PR TITLE
Test driven development

### PR DESCRIPTION
**Pull request by limavfabio**
This PR creates the class Solver that implements the methods factorial, reverse and fizzbuzz, making use of the TDD pattern.

### CODE REVIEW

File: src/spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context '#factorial' do
+    it 'returns 720 for input 6' do
+      expect(@solver.factorial(6)).to eq 720
+    end
+  end
```
Comment (Nemwel-Boniface) [inline src/spec/solver_spec.rb]: - [x] Your factorial tests only factor in one scenario, when the valuse passed in are all positives. What about when the values passed in are negatives or the values passed in is a zero? Kindly include tests of these two edge cases as well.

image

File: src/spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    it 'returns "olleh" for input "hello"' do
+      expect(@solver.reverse('hello')).to eq 'olleh'
+    end
+  end
+
+  context '#fizzbuzz' do
+    it 'returns "1 2 fizz 4 buzz fizz 7 8 fizz buzz 11 fizz 13 14 fizzbuzz" for the first 15 numbers' do
+      expect(@solver.fizzbuzz(15)).to eq '1 2 fizz 4 buzz fizz 7 8 fizz buzz 11 fizz 13 14 fizzbuzz'
+    end
+  end
```
Comment (Nemwel-Boniface) [inline src/spec/solver_spec.rb]: - [x] It would be a good thing to break this down and have at least 4 test cases as provided in the project requirements. eg ```expect(solver.fizzbuzz(3)).to eql('fizz')```

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @limavfabio  and @Camilovelag       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/spec/solver_spec.rb (hunk 1 above)
Comment (Camilovelag) [reply src/spec/solver_spec.rb]: Done! 😄

File: src/spec/solver_spec.rb (hunk 2 above)
Comment (Camilovelag) [reply src/spec/solver_spec.rb]: Done! 😄

**Review by aimalamiri — APPROVED** [review]
Hi @limavfabio, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

ApprovedDavieGIF


## Highlights

- The test are writen very well and works fantastic ✔️ ⭐
- Good job for using the right Git Flow 💯
- Awesome file organization 🗃️
- Commits are descriptive and meaningful 👍
- Linters are properly set up and passing 🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
